### PR TITLE
Corpus backup: overwrite public.zip every time we update the public backup.

### PR DESCRIPTION
**Pull request by Dor1s**
Needed for CIFuzz, and I think it general we should document it in OSS-Fuzz. I don't think there is a way to list a bucket without authentication, therefore it's more of a necessity for sharing OSS-Fuzz public corpora more explicitly.

### CODE REVIEW

File: src/appengine/handlers/cron/corpus_backup.py [hunk 1]
```diff
@@ -66,6 +67,24 @@ def _make_corpus_backup_public(target, corpus_fuzzer_name_override,
...
+      corpus_manager.PUBLIC_BACKUP_TIMESTAMP + os.extsep +
+      corpus_manager.BACKUP_ARCHIVE_FORMAT)
+  public_url = os.path.join(os.path.dirname(corpus_backup_url), filename)
+
+  try:
+    result = storage.copy_blob(corpus_backup_url, public_url)
+    if result:
+      result = storage.set_acl(public_url, 'allUsers')
+  except:
+    result = None
```
Comment (oliverchang) [inline src/appengine/handlers/cron/corpus_backup.py]: Why not log_error here instead? That way we log any exception stacks and can avoid this extraneous `result = None` setting and `if not result`.

Comment (Dor1s) [reply src/appengine/handlers/cron/corpus_backup.py]: You're right, it's pointless. I've just blindly followed the existing pattern. Fixed this in my diff and two other places above. Thank you!

Comment (Dor1s) [reply src/appengine/handlers/cron/corpus_backup.py]: Or actually it had a point, wait a second

Comment (Dor1s) [reply src/appengine/handlers/cron/corpus_backup.py]: it seems the intention was to log an error in both cases:
- if there was an exception
- if `return == False`

Comment (Dor1s) [reply src/appengine/handlers/cron/corpus_backup.py]: Will think more about this tomorrow. Looks like we can get rid of all try-except blocks inside this function and handle an exception outside of it.

Comment (Dor1s) [reply src/appengine/handlers/cron/corpus_backup.py]: yeah, I think now it's much better

**Pull request merged**